Modules/Office365DSC/DSCResources/MSFT_SCSupervisoryReviewPolicy/MSFT_SCSupervisoryReviewPolicy.psm1

```diff
@@ -0,0 +1,189 @@
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter()]
+        [System.String]
+        $Comment,
+
+        [Parameter(Mandatory = $true)]
+        [System.String[]]
+        $Reviewers,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Getting configuration of SupervisoryReviewPolicy for $Name"
+
+    Write-Verbose -Message "Calling Test-SecurityAndComplianceConnection function:"
+    Test-SecurityAndComplianceConnection -GlobalAdminAccount $GlobalAdminAccount
+
+    $PolicyObjects = Get-SupervisoryReviewPolicyV2
+    $PolicyObject = $PolicyObjects | Where-Object {$_.Name -eq $Name}
+
+    if ($null -eq $PolicyObject)
+    {
+        Write-Verbose -Message "SupervisoryReviewPolicy $($Name) does not exist."
+        $result = $PSBoundParameters
+        $result.Ensure = 'Absent'
+        return $result
+    }
+    else
+    {
+        Write-Verbose "Found existing SupervisoryReviewPolicy $($Name)"
+        $result = @{
+            Name               = $PolicyObject.Name
+            Comment            = $PolicyObject.Comment
+            Reviewers          = $PolicyObject.Reviewers
+            Ensure             = 'Present'
+            GlobalAdminAccount = $GlobalAdminAccount
+        }
+
+        Write-Verbose -Message "Found SupervisoryReviewPolicy $($Name)"
+        Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-O365DscHashtableToString -Hashtable $result)"
+        return $result
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter()]
+        [System.String]
+        $Comment,
+
+        [Parameter(Mandatory = $true)]
+        [System.String[]]
+        $Reviewers,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Setting configuration of SupervisoryReviewPolicy for $Name"
+
+    Test-SecurityAndComplianceConnection -GlobalAdminAccount $GlobalAdminAccount
+    $CurrentPolicy = Get-TargetResource @PSBoundParameters
+
+    if (('Present' -eq $Ensure) -and ('Absent' -eq $CurrentPolicy.Ensure))
+    {
+        $CreationParams = $PSBoundParameters
+        $CreationParams.Remove("GlobalAdminAccount")
+        $CreationParams.Remove("Ensure")
+        New-SupervisoryReviewPolicyV2 @CreationParams
+    }
+    elseif (('Present' -eq $Ensure) -and ('Present' -eq $CurrentPolicy.Ensure))
+    {
+        # Easier to delete the Policy and recreate it from scratch;
```

Comment:
Just a thought:
What is the impact of using delete/create instead of updating an existing rule? Isn't recreating triggering a sync of some sort on the background?

Reply:
Good call. Let's play it safe and simply try to update the values of the existing instance using Set- instead.